pgmpy/factors/CPD.py

```diff
@@ -125,6 +127,19 @@ def get_cpd(self):
         else:
             return self.values.reshape(1, np.prod(self.cardinality))
 
+    def __str__(self):
+        return self._str(html=False)
+
+    def _str(self, html=False):
+        table_list = []
+        indexes = np.array(list(product(*[range(i) for i in self.cardinality[1:]])))
+        for i in range(1, len(self.cardinality)):
+            row = self.scope()[i]
+            row = [row]+np.array(self.variables[row])[indexes[:, i-1]].tolist()
```

Comment:
Please don't add list to create a new list. It would always result in creation of new object. Instead use ```row.extend(np.array(self.variables[row])[indexes[:, i-1]].tolist())```.